src/controllers/institution-controller.js

```diff
@@ -19,45 +19,79 @@ export async function fetchProjectInstitutions(req, res) {
 
 export async function addInstitutionToProject(req, res) {
   const projectId = req.params.projectId
+  const name = req.body.name.trim()
   const institutionId = req.body.institutionId
 
-  if (institutionId == null) {
-    res.status(404).json({ message: 'Institution is not found' })
+  if (name == null || name.length == 0) {
+    res.status(404).json({ message: 'Institution cannot be found' })
     return
   }
 
+  let institution = await models.Institution.findByPk(institutionId)
+
   try {
-    const projectInstitution = models.InstitutionsXProject.build({
+    const transaction = await sequelizeConn.transaction()
+
+    if (institution == null) {
+      institution = models.Institution.build({
+        name: name,
+        user_id: req.user.user_id,
+        active: true,
+      })
+
+      await institution.save({
+        transaction: transaction,
+        user: req.user,
+      })
+    }
+
+    const institutionXProject = models.InstitutionsXProject.build({
       project_id: projectId,
-      institution_id: institutionId,
+      institution_id: institution.institution_id,
     })
 
-    const transaction = await sequelizeConn.transaction()
-    await projectInstitution.save({
-      transaction,
+    await institutionXProject.save({
+      transaction: transaction,
       user: req.user,
     })
 
     await transaction.commit()
-    const institution = await models.Institution.findByPk(institutionId)
     res.status(200).json({ institution })
   } catch (e) {
     console.error(e)
-    res.status(500).json({ message: 'could not assign the two' })
+    res
+      .status(500)
+      .json({ message: 'Error adding the institution to the project.' })
   }
 }
 
 export async function removeInstitutionFromProject(req, res) {
   const projectId = req.params.projectId
   const institutionIds = req.body.institutionIds
 
-  if (institutionIds == null) {
+  if (institutionIds == null || !(institutionIds.length > 0)) {
     res.status(404).json({ message: 'Institutions not found' })
     return
   }
 
-  const transaction = await sequelizeConn.transaction()
   try {
+    const dupeInstitutionIds = await models.InstitutionsXProject.findAll({
+      attributes: ['institution_id'],
+      where: {
+        institution_id: institutionIds,
+        project_id: { [Sequelize.Op.not]: projectId },
+      },
+    })
```

Comment:
I just remembered... There is also the InstitutionsXUsers table. You'll have to check that the institution is not in-use by a project and a user before you remove it.